aiida_quantumespresso/calculations/__init__.py

```diff
@@ -618,3 +610,19 @@ def _case_transform_dict(d, dict_name, func_name, transform):
             'are repeated more than once when compared case-insensitively: {}.'
             'This is not allowed.'.format(dict_name, double_keys))
     return new_dict
+
+
+def _pop_parser_options(calc_job_instance, settings_dict):
+    # We need an instance of the parser class to get the parser options key (which is typically 'parser_options')
+    from aiida.plugins import ParserFactory
+    parser_name = calc_job_instance.inputs.get('metadata.options.parser_name', None)
```

Comment:
Good catch. In fact, now that this function is working, I found that my approach is flawed: I can't make an instance of the ParserClass because I would need a CalcJobNode, which I don't have yet. To get around this, I'll turn the methods `get_parser_settings_key()` of `NebParser and `PwParser` into static methods. Alternatively, I could hardcode the 'parser_options' key (also fine by me if you prefer it).